lib/remote/objectqueryhandler.cpp

```diff
@@ -262,23 +262,19 @@ bool ObjectQueryHandler::HandleRequest(
 				continue;
 
 			Type::Ptr reflectionType = joinedObj->GetReflectionType();
-			Expression::Ptr permissionFilter;
-
 			auto it = typePermissions.find(reflectionType.get());
 			bool granted;
 
 			if (it == typePermissions.end()) {
 				String permission = "objects/query/" + reflectionType->GetName();
 
-				Expression *filter = nullptr;
-				granted = FilterUtility::HasPermission(user, permission, &filter);
-				permissionFilter = filter;
+				std::unique_ptr<Expression> permissionFilter;
+				granted = FilterUtility::HasPermission(user, permission, &permissionFilter);
 
-				typePermissions.insert({reflectionType.get(), std::make_pair(granted, permissionFilter)});
-			} else {
-				std::tie(granted, permissionFilter) = it->second;
+				it = typePermissions.insert({reflectionType.get(), std::make_pair(granted, std::move(permissionFilter))}).first;
 			}
 
+			granted = it->second.first;
```

Comment:
I'd still do this like I suggested in https://github.com/Icinga/icinga2/pull/9537#discussion_r997203635, so that it looks like this in the end:

```cpp
if (...) {
    ...

    std::unique_ptr<Expression> permissionFilter;
    bool granted = FilterUtility::HasPermission(...);

    it = typePermissions.insert(...).first;
}

bool granted = it->second.first;
const unique_ptr<Expression>& permissionFilter = it->second.second;
```

This allows you to use a meaningful name below rather than `it->second.second`.